torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,881 @@ namespace fuser {
+  auto transform_exprs = StmtSort::getExprs(
+      ref->fusion(),
+      {ref->getRFactorDomain().begin(), ref->getRFactorDomain().end()});
+
+  // Mapping from rfactor to root, reverse expressions
+  std::reverse(transform_exprs.begin(), transform_exprs.end());
+
+  for (const auto* expr : transform_exprs) {
+    if (const Split* split = dynamic_cast<const Split*>(expr)) {
+      // Initialize state
+      auto find_outer_it = ids.begin();
+      auto outer_pos = ids.size();
+      auto find_inner_it = ids.begin();
+      auto inner_pos = ids.size();
+
+      // Removes all entries to the left of provided `it`, if `it` is not
+      // ids.begin(). Updates all state of finding outer and inner in the ids
+      // vector after erasing.
+      auto clear_left_of = [&find_outer_it,
+                            &outer_pos,
+                            &find_inner_it,
+                            &inner_pos,
+                            &ids,
+                            &split](decltype(find_outer_it) it) {
+        if (it != ids.begin()) {
+          ids.erase(ids.begin(), it);
+        }
+
+        // Set outer it and position
+        find_outer_it = std::find(ids.begin(), ids.end(), split->outer());
+        outer_pos = find_outer_it == ids.end()
+            ? ids.size()
+            : std::distance(ids.begin(), find_outer_it);
+
+        // Set inner it and position
+        find_inner_it = std::find(ids.begin(), ids.end(), split->inner());
+        inner_pos = find_inner_it == ids.end()
+            ? ids.size()
+            : std::distance(ids.begin(), find_inner_it);
+      };
+
+      // Dry run to fill state
+      clear_left_of(ids.begin());
+
+      // Check if the domains out of the split are contiguous in the mapped
+      // domain.
+      if (find_outer_it == ids.end() && find_inner_it != ids.end()) {
+        // Outer dimension was not found, but inner dimension was. Must assume
+        // everything to the left of inner is not contiguously merged.
+        //
+        // Clear left of inner
+        clear_left_of(find_inner_it);
+      } else if (find_outer_it != ids.end() && find_inner_it == ids.end()) {
+        // Inner dimension was not found, outer and anything left of outer are
+        // definitely not contiguous.
+        //
+        // Clear outer and left of outer
+        clear_left_of(find_outer_it + 1);
+        continue;
+      } else if (find_outer_it == ids.end() && find_inner_it == ids.end()) {
+        // Nothing mapped, just continue
+        continue;
+      }
+
+      if (find_outer_it != ids.end() && find_inner_it != ids.end()) {
+        // Both outer and inner mapped.
+        if (outer_pos >= inner_pos) {
+          // Make sure outer is outside inner, otherwise neither could be part
+          // of a continuous mapping. There are cases where we could have
+          // reversible operations e.g.:
+          //    [id{3} id{5} id{6}] -> merge(1, 0)
+          // -> [id{5*3} id{6}] -> split(0, 5)
```

Comment:
I think you meant this?
```suggestion
          // -> [id{5*3} id{6}] -> split(0, 3)
```